### PR TITLE
[dhctl] feat(dhctl-for-commander): add new commander/detach operation, add commander-uuid option for all operations

### PR DESCRIPTION
**Pull request by faroyam**
## Description

- add deckhouse_uuid into commander/attach scan result
- fix incorrect state passed to first on-phase call of bootstrap action
- add commander-uuid consistency checks and commander-uuid storing in the dh cluster
- add commander-uuid stubs for dhctl-server (for testing purposes only)

## Why do we need it, and what problem does it solve?

* new commander/detach operation
  * performs basic check step to ensure cluster terraform state actualization from commander into the target cluster;
  * removes d8-commander-uuid CM;
  * removes commander-agent module from the cluster.

* added optional CommanderUUID for all dhctl operations
  * this is basic safety check preventing single dh cluster from being managed by multiple commanders.
    * once d8-commander-uuid is created for cluster it is not allowed to change commander-uuid to another one without detaching from first commander;
      * d8-commander-uuid CM created in the kube-system namespace;
  * cluster initially bootstrapped in commander-mode with commanderUUID linked with this commander (d8-commander-uuid CM created);
  * cluster which is about to be attached to the commander first checks if it is already linked with another commander and denies attach request if it is;
    * otherwise d8-commander-uuid will be created in the attached cluster on the capture phase.
  * cluster check/converge operations verify that specified commanderUUID is the same as in-cluster;
  * cluster check/converge operations could check existance of d8-commander-uuid and give out-of-sync status, and create d8-commander-uuid if it does not exist respectively:
    * this is for compatibility with clusters created by old versions of commander module: these clusters will be linked to their commander automatically by check+converge;

* all changes are also available for dhctl server.

## What is the expected result?

It is not expected to change behaviour of any dhctl operation not in commander-mode. Commander-uuid should be created only in commander-mode. Commander-uuid should be created only when it is passed (it is currently optional for commander, but will be required in future versions).

```changes
section: dhctl
type: feature
summary: Add new commander/detach operation, add commander-uuid option for all commander operations
impact_level: default
```

### CODE REVIEW

**Review by aleksey-su — CHANGES_REQUESTED** [review]

File: dhctl/pkg/kubernetes/actions/converge/converge.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/google/uuid"
```
[the remark this answers is not in the document]
Comment (faroyam) [reply dhctl/pkg/kubernetes/actions/converge/converge.go]: thanks, now the imports are sorted

File: dhctl/pkg/kubernetes/actions/resources/resources.go [hunk 2]
```diff
@@ -258,3 +241,64 @@ func getUnstructuredName(obj *unstructured.Unstructured) string {
...
+		if namespace == metav1.NamespaceNone && namespaced {
+			namespace = metav1.NamespaceDefault
+		}
+
+		if namespaced {
+			log.InfoF("Deleting %s %s in ns/%s\n", gvr, name, namespace)
+			if err := kubeCl.Dynamic().Resource(gvr).Namespace(namespace).Delete(ctx, name, metav1.DeleteOptions{}); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return fmt.Errorf("unable to remove %s %s: %w", gvr.String(), name, err)
+				}
```
[the remark this answers is not in the document]
Comment (faroyam) [reply dhctl/pkg/kubernetes/actions/resources/resources.go]: added logging and removed code duplication

File: dhctl/pkg/kubernetes/actions/resources/resources.go [hunk 3]
```diff
@@ -258,3 +241,64 @@ func getUnstructuredName(obj *unstructured.Unstructured) string {
...
+			if len(resource.Verbs) == 0 {
+				continue
+			}
+			if resource.Name == name {
+				namespaced = resource.Namespaced
+				break
+			}
+		}
+	}
+	return namespaced, nil
```
[the remark this answers is not in the document]
Comment (faroyam) [reply dhctl/pkg/kubernetes/actions/resources/resources.go]: fixed

**Review by aleksey-su — APPROVED** [review]

File: dhctl/pkg/kubernetes/actions/resources/resources.go [hunk 4]
```diff
@@ -149,25 +151,7 @@ func (c *Creator) TryToCreate() error {
...
-				continue
-			}
-			if resource.Name == name {
-				namespaced = resource.Namespaced
-				break
-			}
-		}
-	}
-	return namespaced, nil
+	return isNamespaced(c.kubeCl, gvk, name)
```
Comment (name212) [inline dhctl/pkg/kubernetes/actions/resources/resources.go:154]: better rename function. In the first time I thought that is recursion method

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [8740](https://github.com/deckhouse/deckhouse/pull/8740) to the branch [release-1.61](https://github.com/deckhouse/deckhouse/tree/release-1.61) successful!

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 9592270232 9592270232-1-con-1-25 /sys/deckhouse-oss/install:pr8607 centos@35.157.240.156

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 9584554364 9584554364-1-con-1-25 /sys/deckhouse-oss/install:pr8607 redos@158.160.101.239

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Pull request merged**